docs/language-server/protocol-project-manager.md

```diff
@@ -185,10 +230,34 @@ This message requests the creation of a new project.
 
 ```typescript
 interface ProjectCreateRequest {
+  /** Name of the project to create. */
   name: String;
+
+  /** Enso Engine version to use for the project.
+   *
+   * Possible values are:
+   * - a semver version string identifying an Enso engine version,
+   * - `default` to use the current default
+   * - `latest-installed` to use the most recent version that is installed on
+   *   the system.
+   */
+  version: String;
+
+  /** Specifies whether to install missing components.
+   *
+   * If not provided, defaults to false.
+   */
+  installMissingComponents?: bool;
 }
 ```
 
+> TODO [RW] discuss: Do we also want a 'latest-available' that would check the
+> repository for newer versions? While we are at it, as the version needs to be
+> specified somehow, we may want to be able to query all available version
+> strings from the repository? To allow users to choose a version for their new
+> project, besides just `engine/list` which only lists currently installed
+> versions.
+
```

Comment:
My proposition is to make engine/list return also possible-to-install versions with appropriate mark.

Reply:
I'd suggest having a separate endpoint, `engine/list-available` (and possibly renaming this one to `engine/list-installed`), as for example if the user is offline they may still want to list the installed engines to uninstall one of them, but listing available versions will in such case fail with a network error. So I think keeping them separate will be simpler.

Reply:
Also I understand that in one way or another the IDE can get a list of all available versions, the `latest-available` option for `version` in `project/create` is not needed because it can be inferred from this list.

In the same way, maybe the `latest-instaled` should also be removed since we can list the installed version, leaving just a specific semver string or missing the field to indicate the default version?

Reply:
> Also I understand that in one way or another the IDE can get a list of all available versions, the latest-available option for version in project/create is not needed because it can be inferred from this list.
> 
> In the same way, maybe the latest-instaled should also be removed since we can list the installed version, leaving just a specific semver string or missing the field to indicate the default version?

Correct

> I'd suggest having a separate endpoint, engine/list-available (and possibly renaming this one to engine/list-installed), as for example if the user is offline they may still want to list the installed engines to uninstall one of them, but listing available versions will in such case fail with a network error. So I think keeping them separate will be simpler.

I'm ok with that.